src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -352,16 +352,11 @@ public async Task<ResourceWrapper> UpdateSearchParameterIndicesAsync(ResourceWra
                 switch (exception.GetSubStatusCode())
                 {
                     case HttpStatusCode.PreconditionFailed:
+                        _logger.LogError(string.Format(Core.Resources.ResourceVersionConflict, weakETag));
                         throw new PreconditionFailedException(string.Format(Core.Resources.ResourceVersionConflict, weakETag));
 
-                    case HttpStatusCode.NotFound:
```

Comment:
Why was the .NotFound case removed? Is this not a possible outcome any longer?

Reply:
Good question! I discovered that this scenario is not possible while attempting (and failing 😄) to write a test that would trigger the `NotFound` case.

The only situation that could lead to a resource not being found when updating its indices would be if the resource was deleted since the operation was kicked off. In this situation, a new deleted version of the resource would be created, and the `PreconditionFailed` case would catch this.